pkg/catalog/catalog.go

```diff
@@ -222,6 +222,7 @@ type Config struct {
 }
 
 type Catalog struct {
+	Cfg                   Config
```

Comment:
Seems like only `StorageConfig` is used by the Catalog - 
Any reason that the full `Config` is needed here (instead of just `StorageConfig`)?

Reply:
Any reason why not?
Does it add any complexity to the code? Added memory usage?

Reply:
So that if someone will ever want to use it only with `StorageConfig`, it'll be easy, and they won't have to check this code and find out that the full `Config` isn't needed, `StorageConfig`.

Reply:
What do you mean by someone wants to use it only with StorageConfig?
Catalog uses the entire config to get config params when initialized.